devito/interfaces.py

```diff
@@ -501,145 +588,19 @@ def initialize(self):
-    def dyr(self):
-        """Symbol for the derivative wrt to y with a right stencil"""
-        return first_derivative(self, order=self.space_order, dim=y, side=right)
-
-    @property
-    def dzl(self):
-        """Symbol for the derivative wrt to z with a left stencil"""
-        return first_derivative(self, order=self.space_order, dim=z, side=left)
-
-    @property
-    def dzr(self):
-        """Symbol for the derivative wrt to z with a right stencil"""
-        return first_derivative(self, order=self.space_order, dim=z, side=right)
-
-    @property
-    def dx2(self):
-        """Symbol for the second derivative wrt the x dimension"""
-        width_h = int(self.space_order/2)
-        indx = [(x + i * x.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(x, x).as_finite_difference(indx)
-
-    @property
-    def dy2(self):
-        """Symbol for the second derivative wrt the y dimension"""
-        width_h = int(self.space_order/2)
-        indy = [(y + i * y.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(y, y).as_finite_difference(indy)
-
-    @property
-    def dz2(self):
-        """Symbol for the second derivative wrt the z dimension"""
-        width_h = int(self.space_order/2)
-        indz = [(z + i * z.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(z, z).as_finite_difference(indz)
-
-    @property
-    def dx2y2(self):
-        """Symbol for the second cross derivative wrt the x,y dimension"""
-        return second_derivative(self.dx2, dim=y, order=self.space_order)
-
-    @property
-    def dx2z2(self):
-        """Symbol for the second cross derivative wrt the x,z dimension"""
-        return second_derivative(self.dx2, dim=z, order=self.space_order)
-
-    @property
-    def dy2z2(self):
-        """Symbol for the second cross derivative wrt the y,z dimension"""
-        return second_derivative(self.dy2, dim=z, order=self.space_order)
-
-    @property
-    def dx4(self):
-        """Symbol for the fourth derivative wrt the x dimension"""
-        width_h = max(int(self.space_order / 2), 2)
-        indx = [(x + i * x.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(x, x, x, x).as_finite_difference(indx)
-
-    @property
-    def dy4(self):
-        """Symbol for the fourth derivative wrt the y dimension"""
-        width_h = max(int(self.space_order / 2), 2)
-        indy = [(y + i * y.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(y, y, y, y).as_finite_difference(indy)
-
-    @property
-    def dz4(self):
-        """Symbol for the fourth derivative wrt the z dimension"""
-        width_h = max(int(self.space_order / 2), 2)
-        indz = [(z + i * z.spacing) for i in range(-width_h, width_h + 1)]
-
-        return self.diff(z, z, z, z).as_finite_difference(indz)
-
-    @property
     def laplace(self):
         """Symbol for the second derivative wrt all spatial dimensions"""
```

Comment:
not part of this PR, I know, but since we're fixing this stuff... the docstring, again: this returns something more than "just a Symbol"

Reply:
Agreed, it's actually a symbolic expression that's returned.